c10/xpu/impl/XPUGuardImpl.h

```diff
@@ -84,7 +95,12 @@ struct XPUGuardImpl final : public c10::impl::DeviceGuardImplInterface {
 
     auto* xpu_event = reinterpret_cast<sycl::event*>(*event);
     const XPUStream xpu_stream{stream};
-    *xpu_event = xpu_stream.queue().ext_oneapi_submit_barrier();
+
+    // Delete the event previously recorded.
+    if (xpu_event)
+      delete xpu_event;
```

Comment:
Would this still work if some stream is waiting on this version of event while a new record is issued? Do we have UT to cover this?

Reply:
It works. Rich the UT to cover this scenario.